src/beamlime/applications/handlers.py

```diff
@@ -153,9 +162,18 @@ def __init__(self, workflow: StatelessWorkflow) -> None:
         super().__init__()
 
     def reduce_data(self, message: DataReady) -> WorkflowResultUpdate:
-        content = JSONGroup(message.content)
+        nxevent_data = {
+            key: JSONGroup(value) for key, value in message.content.nxevent_data.items()
+        }
+        nxlog = {key: JSONGroup(value) for key, value in message.content.nxlog.items()}
         self.info("Running data reduction")
-        return WorkflowResultUpdate(content=self.workflow(content))
+        return WorkflowResultUpdate(
+            content=self.workflow(
+                nexus_filename=message.content.nexus_filename,
+                nxevent_data=nxevent_data,
+                nxlog=nxlog,
+            )
+        )
```

Comment:
This part will be also re-written since we'll not have one big `_store` anymore, and keep stream-module-specific stores.